docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -987,6 +988,34 @@ The `-fexceptions` option is also needed for C code that needs to interoperate w
 
 ---
 
+## Enable pre-determined set of hardening options in GCC
+
+| Compiler Flag                             | Supported since | Description                                                         |
+|:----------------------------------------- |:---------------:|:------------------------------------------------------------------- |
+| <span id="-fhardened">`-fhardened`</span> | GCC 14.0.0      | Enable pre-determined set of hardening options for C and C++ in GCC |
+
+The `-fhardened` umbrella option enables a pre-determined set of hardening options for C and C++ on GNU/Linux targets[^gcc-fhardened]. The precise set of foptions may change between major releases of GCC. The exact set of options for a specific GCC version can be displayed using the `--help=hardened` option.
+
+In GCC 14, the set of options are:
+
+|:--------------------------------------------------------------- |
+| [`-D_FORTIFY_SOURCE=3`](#-D_FORTIFY_SOURCE=3)                   |
+| [`-D_GLIBCXX_ASSERTIONS`](#-D_GLIBCXX_ASSERTIONS)               |
+| [`-ftrivial-auto-var-init=zero`](#-ftrivial-auto-var-init) |
+| [`-fPIE -pie`](#-fPIE_-pie)                                     |
+| [`-Wl,-z,relro,-z,now`]((#-Wl,-z,relro))                        |
+| [`-fstack-protector-strong`](#-fstack-protector-strong)         |
+| [`-fstack-clash-protection`](#-fstack-clash-protection)         |
+| [`-fcf-protection=full`](#-fcf-protection=full)   (x86_64 only) |
+
+### Additional Considerations
+
+Options explicitly specified on the compiler command line always take precedence, e.g., `-fhardened -fstack-protector` will enable `-fstack-protector`, not `-fstack-protector-strong` implied by `-fhardened`.
```

Comment:
There's `-Whardened`, which should warn if a flag ends up downgrading or disabling hardening enabled by `-fhardened`. It should get enabled whenever `-fhardened` is specified.

Reply:
Thanks, that's a very useful addition. I'll add it to the table in this section as well.